ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/metadata/MetadataPresenterImpl.java

```diff
@@ -217,6 +222,16 @@ public static Avu newMetadata(String attr, String value, String unit) {
         return avu;
     }
 
+    @Override
+    public Avu  setAvuModelKey(Avu avu) {
+        if (avu != null) {
+            final AutoBean<Avu> avuBean = AutoBeanUtils.getAutoBean(avu);
+            avuBean.setTag(AVU_BEAN_TAG_MODEL_KEY, String.valueOf(unique_avu_id++));
+            return avuBean.as();
```

Comment:
I guess I liked that the liststore's ModelKeyProvider would handle anything related to keys so you don't have to worry about it anywhere else in the code.  But your explanation makes sense too. 👍